test/suites/dev/test-xcm-v2/test-mock-hrmp-transact-ethereum-1.ts

```diff
@@ -0,0 +1,156 @@
+        (await context.polkadotJs().query.system.account(descendOriginAddress)) as any
+      ).data.free.toBigInt();
+      expect(balance).to.eq(transferredBalance);
+    });
+
+    it({
+      id: "T01",
+      title: "should receive transact and should be able to execute",
+      test: async function () {
+        // Get Pallet balances index
+        const metadata = await context.polkadotJs().rpc.state.getMetadata();
+        const balancesPalletIndex = metadata.asLatest.pallets
+          .find(({ name }) => name.toString() == "Balances")!
+          .index.toNumber();
+
+        const amountToTransfer = transferredBalance / 10n;
+
+        const xcmTransactions = [
+          {
+            V1: {
+              gas_limit: 21000,
+              fee_payment: {
+                Auto: {
+                  Low: null,
+                },
+              },
+              action: {
+                Call: random.address,
+              },
+              value: amountToTransfer,
+              input: [],
+              access_list: null,
+            },
+          },
+          {
+            V2: {
+              gas_limit: 21000,
+              action: {
+                Call: random.address,
+              },
+              value: amountToTransfer,
+              input: [],
+              access_list: null,
+            },
+          },
+        ];
+
+        let expectedTransferredAmount = 0n;
+        let expectedTransferredAmountPlusFees = 0n;
+
+        const targetXcmWeight = 1_325_000_000n + 25_000_000n;
+        const targetXcmFee = targetXcmWeight * 50_000n;
+
+        for (const xcmTransaction of xcmTransactions) {
+          expectedTransferredAmount += amountToTransfer;
+          expectedTransferredAmountPlusFees += amountToTransfer + targetXcmFee;
+          // TODO need to update lookup types for xcm ethereum transaction V2
+          const transferCall = context.polkadotJs().tx.ethereumXcm.transact(xcmTransaction as any);
+          const transferCallEncoded = transferCall?.method.toHex();
+
+          // We are going to test that we can receive a transact operation from parachain 1
+          // using descendOrigin first
+          const xcmMessage = new XcmFragment({
+            assets: [
+              {
+                multilocation: {
+                  parents: 0,
+                  interior: {
+                    X1: { PalletInstance: balancesPalletIndex },
+                  },
+                },
+                fungible: targetXcmFee,
+              },
+            ],
+            weight_limit: new BN(targetXcmWeight.toString()),
+            descend_origin: sendingAddress,
+          })
+            .descend_origin()
+            .withdraw_asset()
+            .buy_execution()
+            .push_any({
+              Transact: {
+                originType: "SovereignAccount",
+                // 21_000 gas limit + db read
+                requireWeightAtMost: new BN(525_000_000).add(new BN(25_000_000)),
```

Comment:
this is what's so gross about `BN()`. bigints are good enough for normally all integer maths now.

BN() has actually been taken out of Ethers.js and Web3.js, both use bigint